server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -122,18 +122,16 @@ private void getMultipleReposSnapshotInfo(@Nullable SnapshotsInProgress snapshot
                             return new GetSnapshotsResponse(responses);
                         }), repos.size());
 
-        // run concurrently for all repos on GENERIC thread pool
         for (final RepositoryMetadata repo : repos) {
             final String repoName = repo.name();
-            threadPool.generic().execute(ActionRunnable.wrap(
-                ActionListener.delegateResponse(groupedActionListener, (groupedListener, e) -> {
-                    if (e instanceof ElasticsearchException) {
-                        groupedListener.onResponse(GetSnapshotsResponse.Response.error(repoName, (ElasticsearchException) e));
-                    } else {
-                        groupedListener.onFailure(e);
-                    }
-                }), wrappedListener -> getSingleRepoSnapshotInfo(snapshotsInProgress, repoName, snapshots, ignoreUnavailable, verbose,
-                    ActionListener.map(wrappedListener, snInfos -> GetSnapshotsResponse.Response.snapshots(repoName, snInfos)))));
+            getSingleRepoSnapshotInfo(snapshotsInProgress, repoName, snapshots, ignoreUnavailable, verbose, ActionListener.map(
```

Comment:
No need to fork for this here already, we might just be loading `_current` and never do any IO.